pytest/frontend/test_parameters.py

```diff
@@ -0,0 +1,125 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Integration tests to make sure hbar values set correctly in returned results"""
+import pytest
+
+pytestmark = pytest.mark.frontend
+
+import numpy as np
+import tensorflow as tf
+
+import strawberryfields as sf
+from strawberryfields import ops
+from strawberryfields.parameters import Parameter
+
+# make test deterministic
+np.random.random(32)
+
+test_values = [
+    3,
+    0.14,
+    4.2 + 0.5j,
+    np.random.random(3),
+    tf.Variable(2),
+    tf.Variable(0.4),
+    tf.Variable(0.8 + 1.1j),
+]
+
+test_parameters = [Parameter(i) for i in test_values]
+
+
+def test_parameter_wrapping_integer():
+    """Tests that ensure wrapping works with integers"""
+    var = 5
+    res = Parameter._wrap(var)
+    assert isinstance(res, Parameter)
+    assert res.x == var
+
+
+def test_parameter_wrapping_parameters():
+    """Tests that ensure wrapping works with other parameters"""
+    var = 5
+    var = Parameter(var)
+    res = Parameter._wrap(var)
+    assert isinstance(res, Parameter)
+    assert res.x == var
+
+
+def test_parameter_no_shape():
+    """Tests that ensure wrapping occurs as expected"""
+    var = Parameter(5)
+    res = var.shape
+    assert res is None
+
+
+def test_parameter_shape():
+    """Tests that ensure wrapping occurs as expected"""
+    var = np.array([[1, 2, 3], [4, 5, 6]])
+    p = Parameter(var)
+    res = p.shape
+    assert res == (2, 3)
+
+
+def test_parameter_shape_tf():
+    """Tests that ensure wrapping occurs as expected"""
+    var = np.array([[1, 2, 3], [4, 5, 6]])
+    p = Parameter(tf.convert_to_tensor(var))
+    res = p.shape
+    assert res == (2, 3)
+
+
+@pytest.mark.parametrize("p", test_parameters)
+@pytest.mark.parametrize("q", test_parameters)
+def test_parameter_arithmetic(p, q):
+    """Test parameter addition works as expected"""
+    # TODO: make sure values are actually adding/etc together!
```

Comment:
Yep. I noticed that we have tests to check if arithmetic between two parameter classes creates a parameter class. But no assert statements to check if the arithmetic gave the right value!

Reply:
should we add the new asserts then?

Reply:
Can we do that in a new issue/PR? Since the scope of this PR is already quite large. I tried not to change any of the tests I ported over, apart from just moving from unittest->pytest constructs